From 3acfc32fb2c82698b80773df3ca964b40af9e2d4 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Fri, 20 Dec 2019 11:57:00 -0800
Subject: [PATCH] audio-hal: fix audioserver crash

PD#SWPL-17555

Problem:
audio server crashes when we use tvclient

Solution:
return if null ptr

Verify:
Local

Change-Id: I6ef87af9bb402da22d1996ddd883f6ff4807f388
---
 audio_hal/sub_mixing_factory.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/audio_hal/sub_mixing_factory.c b/audio_hal/sub_mixing_factory.c
index 59843cf..628a4c3 100644
--- a/audio_hal/sub_mixing_factory.c
+++ b/audio_hal/sub_mixing_factory.c
@@ -1581,6 +1581,11 @@ static int out_flush_subMixingPCM(struct audio_stream_out *stream)
 int switchNormalStream(struct aml_stream_out *aml_out, bool on)
 {
     ALOGI("+%s() stream %p, on = %d", __func__, aml_out, on);
+    if (!aml_out) {
+        ALOGE("%s(), no stream", __func__);
+        return -EINVAL;
+
+    }
     if (!aml_out->is_normal_pcm) {
         ALOGE("%s(), not normal pcm stream", __func__);
         return -EINVAL;
-- 
2.7.4

